### PR TITLE
[RAPTOR-5935] Adding better explanation for estimators, pipelines, an…

### PR DESCRIPTION
**Pull request by drenfr01**
…d transforms

# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary

Rename the folders of the task_templates to appear in sequential order (1_transforms, 2_estimators, 3_pipelines). Then update all of the test artifacts that depend on these

## Rationale

We want to ensure users have a great experience and can get started quickly from the github repo. It's not apparent immediately that transforms are usually the simplest custom tasks, then estimators, and the finally pipelines as the most complex. So to emphasize that we're ordering the task templates folders in order of complexity in both the README and how they appear in the Github file structure

### CODE REVIEW

File: task_templates/README.md [hunk 1]
```diff
@@ -1,14 +1,55 @@
 # Templates for Custom Tasks
 
-In this directory are three folders - estimators, transforms, and pipelines
+In this directory there are three folders - estimators, transforms, and pipelines. 
+These are collectively known as "tasks". Each of these folders
+contain code snippets, i.e. tasks, that can be uploaded to DataRobot and incorporated into a blueprint.
```
Comment (MikhailYakubovskiy) [inline task_templates/README.md:5]: I agree that it's a good idea to better explain how a task works, in that case we should elaborate a bit more though. Perhaps we can say something like this:
"DataRobot runs task's code as a part of the blueprint when it's being trained or used to score new data. To make sure DataRobot can correctly execute the code, the code must comply with a few requirements: (1) there must be a file custom.py or custom.R (depending on the language you use) and (2) custom.py/custom.R must contain one or multiple hooks (e.g. fit/score/transform/etc) that DataRobot will call and pass data to when a task is being trained or used for scoring. You can use multiple supporting files as well as needed, e.g. to clean up your code with helper functions. Once uploaded each tasks appears as a single box in the Blueprint. Please refer to Composable ML documentation to learn more about how tasks work."

File: task_templates/README.md [hunk 2]
```diff
@@ -1,14 +1,55 @@
...
+can use multiple supporting files as well as needed, e.g. to clean up your code with helper functions.  
+Once uploaded each tasks appears as a single box in the Blueprint. 
+The primary difference between these three are their output and their scope:
 
-* estimators - meant to be used as the prediction step in a blueprint
-* transforms - meant to be used as a preprocessing step in a blueprint
-* pipelines - meant to be added as a single task blueprint, these have preprocessing and an
-  estimator built into a single task
+* transforms - transform the input data, e.g. one hot encoding, numeric scaling, etc. 
+  Their output is always a dataframe to be used by other transforms or estimators.
```
Comment (MikhailYakubovskiy) [inline task_templates/README.md:12]: we could mention here that output also includes headers/column names (it's a common question and many users don't realize that)

File: task_templates/README.md [hunk 3]
```diff
@@ -1,14 +1,55 @@
...
+* estimators - predict (i.e. estimate) a new value using the input data, e.g. Logistic Regression or SVM. 
+  In DataRobot the final task, i.e. box, in any Blueprint must be an estimator. Note: a single Blueprint
+  can, and often does, contain multiple estimators.
+* pipelines - allow a user to create a single task, i.e. box, in the Blueprint that incorporates
+multiple transforms and/or estimators. This is useful if a user has a fully developed model pipeline
```
Comment (MikhailYakubovskiy) [inline task_templates/README.md:17]: lets not call them "users" in the docs

File: task_templates/README.md (hunk 1 above)
Comment (drenfr01) [reply task_templates/README.md:5]: done!

**Comment by drenfr01** [comment]
jarvis test cmrunner_integration

File: task_templates/README.md [hunk 4]
```diff
@@ -81,4 +136,4 @@ Except for where noted, our pipelines are verified within our functional test fr
 15. `python3_keras_vizai_joblib` - Trains a keras model on base64 encoded images.
 
 There is also a growing repository of re-usable tasks that might address your use case. You can find
-it here: https://github.com/datarobot-community/custom-models/tree/master/custom_tasks
\ No newline at end of file
+it here: htt****ps://github.com/datarobot-community/custom-models/tree/master/custom_tasks
```
Comment (scottp-ml) [inline task_templates/README.md:139]: what is going on here with htt****ps?

File: tests/drum/constants.py [hunk 5]
```diff
@@ -13,9 +13,9 @@
 TESTS_ARTIFACTS_PATH = os.path.join(TESTS_FIXTURES_PATH, "drop_in_model_artifacts")
 TESTS_DATA_PATH = os.path.join(TESTS_ROOT_PATH, "testdata")
 TESTS_DEPLOYMENT_CONFIG_PATH = os.path.join(TESTS_DATA_PATH, "deployment_config")
-TRAINING_TEMPLATES_PATH = os.path.join(TESTS_ROOT_PATH, "..", "task_templates", "pipelines")
-TRANSFORM_TEMPLATES_PATH = os.path.join(TESTS_ROOT_PATH, "..", "task_templates", "transforms")
-ESTIMATORS_TEMPLATES_PATH = os.path.join(TESTS_ROOT_PATH, "..", "task_templates", "estimators")
+TRAINING_TEMPLATES_PATH = os.path.join(TESTS_ROOT_PATH, "..", "task_templates", "3_pipelines")
```
Comment (scottp-ml) [inline tests/drum/constants.py:16]: consider putting them in the order they are found in the directory

**Review by scottp-ml — APPROVED** [review]
overall good. couple of comments/suggestions

File: custom_model_runner/README.md [hunk 6]
```diff
@@ -308,7 +308,7 @@ adding the training model into DataRobot.
 You can try this out on our sklearn classifier model template this this command:
 
 ```
-drum fit --code-dir task_templates/pipelines/python3_sklearn_binary --target-type binary --target Species --input \
+drum fit --code-dir task_templates/3_pipelines/python3_sklearn_binary --target-type binary --target Species --input \
```
Comment (lkanggithub) [inline custom_model_runner/README.md:311]: A little bit context included in the PR description would be helpful. What is this renaming?

**Review by eric-s-s — APPROVED** [review]

File: custom_model_runner/README.md (hunk 6 above)
Comment (drenfr01) [reply custom_model_runner/README.md:311]: Sorry, that Rationale and Summary was not very helpful. I added a much longer summary & rationale above

**Review by lkanggithub — APPROVED** [review]
LGTM

**Pull request merged**